Codeinwp/neve
# Menu color issues

### Summary

- In this PR, I improved the way highlighting the active menu item works by adding the same class for the active menu item and the active ancestor. This way, I could target the active menu item and the active ancestor with just one shorter class.
- I removed the `a:not[href=*#]` selector after doing some research on other themes. The sector prevented the active color from applying to elements that had "#" inside their links.
- Before this PR, the caret was not taking the color of the active menu item, so I fixed that too.
- I added a new color control for the second skin mode for the secondary menu since the primary menu had it but not the secondary menu 


### Will affect the visual aspect of the product
YES

### Screenshots 
https://vertis.d.pr/i/MKGkuy

### Test instructions
- Create a menu and link it to a menu component. Make sure the menu has sub-items.
- Add the menu component inside the header and check its behavior on frontend
- The ancestor of a menu should have the active color applied
- The caret should also have the active color applied
- This should apply no matter the menu item level. Eg. if you have a level 4 menu ( menu in menu x 4 ) each ancestor should have the active color
- Check that the hover color applies to the active menu item


## Check before Pull Request is ready:



Closes #3847, #3913.
